sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -136,9 +136,13 @@ public ReactorConnection(String connectionId, ConnectionOptions connectionOption
                 final Mono<AmqpEndpointState> activeEndpoint = getEndpointStates()
                     .filter(state -> state == AmqpEndpointState.ACTIVE)
                     .next()
-                    .timeout(operationTimeout, Mono.error(() -> new AmqpException(true, TIMEOUT_ERROR,
-                        String.format("Connection '%s' not active within the timout: %s.", connectionId, operationTimeout),
-                        handler.getErrorContext())));
+                    .timeout(operationTimeout, Mono.error(() -> {
+                        AmqpException exception = new AmqpException(true, TIMEOUT_ERROR,
+                            String.format("Connection '%s' not active within the timout: %s.", connectionId, operationTimeout),
+                            handler.getErrorContext());
+                        this.handler.onError(exception);
```

Comment:
LMK if it's the right thing to do and/or if there is an easier way to let AmqpChannelProcessor to retry

Reply:
Since our idea here is to signal timeout to the subscription that AmqpChannelProcessor made to the endpoint-states, I wonder we can scope required logic to "AmqpChannelProcessor::onNext", as in –

```java
connectionSubscription = endpointStatesFunction.apply(amqpChannel)
    .filter(state -> state == AmqpEndpointState.ACTIVE)
   .next()
   .doOnNext(__ -> {
       retryAttempts.set(0);
       logger.info("Channel is now active.");
   });
   .timeout(…) // timeout here
  .subscribe(..
```

The reason is – currently there is a chance of non-serialized access, since it opens a potential for handler.onError to be called concurrently from two different threads (timeout thread on timeout, Qpid thread on wire error). By moving the timeout to AmqpChannelProcessor::onNext, we can continue to expect that only wire error from Qpid Thread signaled through handler.onError, that guarantees serialized signaling. Please share your thoughts on this apporach.

Reply:
Looking more into it, I see today in case of non-serialized access, we let the error emission to retry, given that retry going to see already errored state, the retry will get ignored. Seems we are good with your proposed approach of calling handler.onError from Connection type.

Reply:
Only feedback I’ve is to put the onError call inside ``` if (!isV2) ``` block, so it won’t be applied for v2 case, given timeout is taken care by ReactorConnectionCache.

Reply:
thanks for checking and done